rollup/sync_service/bridge_client.go

```diff
@@ -0,0 +1,170 @@
+package sync_service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum"
+	"github.com/scroll-tech/go-ethereum/accounts/abi"
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/rpc"
+)
+
+// BridgeClient is a wrapper around EthClient that adds
+// methods for conveniently collecting L1 messages.
+type BridgeClient struct {
+	client                EthClient
+	confirmations         rpc.BlockNumber
+	l1MessageQueueAddress common.Address
+}
+
+func newBridgeClient(ctx context.Context, l1Client EthClient, l1ChainId uint64, confirmations rpc.BlockNumber, l1MessageQueueAddress *common.Address) (*BridgeClient, error) {
+	if l1MessageQueueAddress == nil {
+		return nil, errors.New("must pass l1MessageQueueAddress to BridgeClient")
+	}
+
+	// sanity check: compare chain IDs
+	got, err := l1Client.ChainID(ctx)
```

Comment:
check if `l1Client` is `nil`?

Reply:
This is checked in `SyncService` before instantiating `BridgeClient`.